src/controllers/ioSignController.ts

```diff
@@ -126,6 +135,93 @@ export default class IoSignController {
       )()
     );
 
+  /**
+   * Create a Signature from a Signature Request
+   */
+  public readonly createSignature = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorValidation
+    | IResponseErrorNotFound
+    | IResponseSuccessJson<SignatureDetailView>
+  > =>
+    withUserFromRequest(req, async user =>
+      pipe(
+        req.body,
+        CreateSignatureBody.decode,
+        E.mapLeft(responseErrorValidation),
+        TE.fromEither,
+        TE.chainW(body =>
+          pipe(
+            sequenceS(TE.ApplySeq)({
+              signerId: pipe(
+                retrieveSignerId(this.ioSignService, user.fiscal_code),
+                TE.mapLeft(
+                  responseErrorInternal(
+                    "Error retrieving the signer id for this users"
+                  )
+                )
+              ),
+              userProfile: pipe(
+                profileWithEmailValidatedOrError(this.profileService, user),
+                TE.mapLeft(
+                  responseErrorInternal(
+                    "Error retrieving a user profile with validated email address"
+                  )
+                )
+              )
+            }),
+            TE.map(({ userProfile, signerId }) =>
+              this.ioSignService.createSignature(
```

Comment:
You're right! The `ioSignService` methods could return `TaskEither` instead of `Promise` and could be composed via `chain` instead of `map` (or maybe we can wrap these methods in a `TE.tryCatch`) but since this is an effectful controller (that lift the `TaskEither` to a `Task` first and then immediately execute it!) using a `TE.map` or a `TE.chain` it's the same.

We decided that since the `map` it's also used on other controllers is already merged, we'll refactor `ioSignService` class to be entirely pure in a later PR.